agent/api/ecsclient/client.go

```diff
@@ -164,18 +164,30 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 
 	resources, err := client.getResources()
 	if err != nil {
-		return "", err
+		return "","", err
 	}
 
 	registerRequest.TotalResources = resources
 	resp, err := client.standardClient.RegisterContainerInstance(&registerRequest)
 	if err != nil {
 		seelog.Errorf("Unable to register as a container instance with ECS: %v", err)
-		return "", err
+		return "","", err
+	}
+
+	var availabilityzone_name = "ecs.availability-zone"
+	var availabilityzone = ""
+	attrs := resp.ContainerInstance.Attributes
+	if(resp != nil){
+		for _, attr := range attrs {
+			if aws.StringValue(attr.Name) == availabilityzone_name {
+				availabilityzone = aws.StringValue(attr.Value)
+			}
+		}
 	}
+
 	seelog.Info("Registered container instance with cluster!")
 	err = validateRegisteredAttributes(registerRequest.Attributes, resp.ContainerInstance.Attributes)
-	return aws.StringValue(resp.ContainerInstance.ContainerInstanceArn), err
+	return aws.StringValue(resp.ContainerInstance.ContainerInstanceArn), availabilityzone,err
```

Comment:
nit: space, this happens a lot, have you configured `gofmt` on your editor? It will automatically adjust spaces.